app/test_server.py

```diff
@@ -24,20 +16,75 @@ def test_true():
     assert True, "We have a problem!"
 
 
-@pytest.mark.xfail(PYTHON3 or not TRAVIS_CI, reason=REASON)
+@pytest.mark.xfail(reason='small_test() needs to be fixed.')
+def test_small_test():
+    small_test()
+
+
+@pytest.mark.xfail
 def test_invalid_url_api_call():
     response = requests.get('http://localhost:7001/api/v1/search/invalid_url')
     assert response.json()['Status Code'] == 404
+    assert response.status_code == response.json()['Status Code']
 
 
 def make_engine_api_call(engine_name):
     url = 'http://localhost:7001/api/v1/search/' + engine_name
     assert requests.get(url).json()['Status Code'] == 400, engine_name
 
+    if engine_name in ('exalead', 'mojeek'):
```

Comment:
Not testing for these?

Reply:
I will add a comment to explain.  Thanks for your review.